src/main/scala/com/tresata/spark/kafka/KafkaRDD.scala

```diff
@@ -192,7 +192,7 @@ class KafkaRDD private (sc: SparkContext, val topic: String, val offsets: Map[In
     "missing offsets for partition(s) " + (leaders.keySet -- offsets.keySet).toList.sorted.mkString(", ")
   )
 
-  protected def getPartitions: Array[Partition] = leaders.zipWithIndex.map{ case ((partition, leader), index) =>
+  protected def getPartitions: Array[Partition] = leaders.toSeq.sortBy(_._1).zipWithIndex.map{ case ((partition, leader), index) =>
```

Comment:
does this mean that with this change partition is now always equal to index?
if so do we still need both?